### ISSUE TITLE
Parser bug with command substitutions in strings inside parenthesis

### ISSUE THREAD

**Issue by Darkshadow2** [fish-shell/fish-shell#8500]
Version: 3.3.1-671-ge2331eb03

(Just noticed with this version, most likely is there since this was introduced)

I've been updating some scripts to use this, and ran into this bug.  If you have two (or more) command substitutions inside parenthesis in a string, you get a parser error about mismatched parenthesis.  An example:

```fish
$ math "($(random) - $(random))"
fish: Mismatched parenthesis
math "($(random) - $(random))"
     ^
```

This isn't just a problem with `math`, it's a general problem.  You can replace it with any command, and you'll get the same error.

It does work with only one command substitution inside parenthesis:

```fish
math "($(random) / 2) x $(random)"
```

**Comment by krobelus** [comment]
Thanks for reporting. I have a fix (300746705596e856f6b27677a361b1d98afee03b) but it uses evil magics so I'll try to find a better one

**Comment by zanchey** [comment]
I've tagged this as a regression only because it doesn't need to go in the release notes.
